### PR TITLE
Remove offset + fix top result

### PR DESCRIPTION
**Pull request by AustinGitHub**
Issue - https://github.com/sigma67/ytmusicapi/issues/518
These offsets cause an issue 

Fix, I test this way 
image

Before
[BeforeChange.txt](https://github.com/sigma67/ytmusicapi/files/13929569/BeforeChange.txt)

After
[AfterChanges.txt](https://github.com/sigma67/ytmusicapi/files/13929572/AfterChanges.txt)

Differences
image

Only issue is this 

image

But this is because code forced in to append `None` for artist without `Id` but then again, what if an artist name is "Album" can't just filter that out by keyword. I think this append of `None` should be taken out because with the way youtube music and youtube topic channels are auto-created I think it is safe to say most artists would have an id, no? 

Let me know and I can commit that change, thanks!

### CODE REVIEW

**Comment by AustinGitHub** [comment]
Tried to run the pyTests but it's very complicated. The oauth works fine, but the browser one just says paste the request headers from firefox and press enter, but what request headers? I am confused on that part. 

Also, can't you write a new script to run both of those functions and then move the file as well? Would make it a lot easier.

**Comment by jcbirdwell** [comment]
> but what request headers?

Request headers come from a post request to a user specific endpoint. So with your network monitor open navigate from something like home -> library and use the post request associated with it.

File: ytmusicapi/parsers/search.py [hunk 1]
```diff
@@ -34,8 +34,17 @@ def parse_top_result(data, search_result_types):
...
 
     if result_type in ["song", "video", "album"]:
+        on_tap = data.get("onTap")
+        if on_tap is not None:
+            if "watchEndpoint" in on_tap:
+                if nav(on_tap, WATCH_VIDEO_ID) is not None:
+                    search_result["videoId"] = nav(on_tap, WATCH_VIDEO_ID)
+            if "navigationEndpoint" in on_tap:
+                if nav(on_tap, NAVIGATION_VIDEO_TYPE) is not None:
+                    search_result["videoType"] = nav(on_tap, NAVIGATION_VIDEO_TYPE)
```
Comment (sigma67) [inline ytmusicapi/parsers/search.py:44]: Instead of triple-if, just navigate directly to the key you want to access using `nav`. Make sure to set `none_if_absent` (i.e. `nav(data, keys, True)`) - this way it'll just error out with `None` at the first missing key in the chain of `keys`

```suggestion
       search_result["videoId"] = nav(data, ["onTap"] + WATCH_VIDEO_ID, True)
       search_result["videoType"] = nav(data, ["onTap"] + NAVIGATION_VIDEO_TYPE, True)
```

File: ytmusicapi/parsers/search.py [hunk 2]
```diff
@@ -34,8 +34,17 @@ def parse_top_result(data, search_result_types):
...
+
         search_result["title"] = nav(data, TITLE_TEXT)
-        runs = nav(data, ["subtitle", "runs"])[2:]
+        runs = nav(data, ["subtitle", "runs"])
```
Comment (sigma67) [inline ytmusicapi/parsers/search.py:47]: why this change?

File: ytmusicapi/parsers/search.py [hunk 3]
```diff
@@ -125,7 +134,7 @@ def parse_search_result(data, search_result_types, result_type, category):
         search_result["duration"] = None
         search_result["year"] = None
         flex_item = get_flex_column_item(data, 1)
-        runs = flex_item["text"]["runs"][default_offset:]
+        runs = flex_item["text"]["runs"]
```
Comment (sigma67) [inline ytmusicapi/parsers/search.py:137]: why this change?

**Review by sigma67 — COMMENTED** [review]
Thanks for the contribution, see comments.

I suggest we discuss the setup issue over in #520. I think we could improve `CONTRIBUTING.md` and I'm open towards a script if it's feasible. Could probably also simplify the `test.cfg` a bit. 

Request headers are documented here: https://ytmusicapi.readthedocs.io/en/stable/setup/browser.html

But I can see how you'd miss that if you never bothered with the old way of doing things since we have oauth now

File: ytmusicapi/parsers/search.py (hunk 1 above)
Comment (AustinGitHub) [reply ytmusicapi/parsers/search.py:44]: Sure I can do that

File: ytmusicapi/parsers/search.py (hunk 3 above)
Comment (AustinGitHub) [reply ytmusicapi/parsers/search.py:137]: Like the other, this was part of the issue too

With my change
image

With original change
image

I crossed check the Json files I am writing too and I see with removing this offset it has better chance of showing the artist name with Id over nothing.

File: ytmusicapi/parsers/search.py (hunk 1 above)
Comment (AustinGitHub) [reply ytmusicapi/parsers/search.py]: I made this change, thank you!

File: ytmusicapi/parsers/search.py (hunk 3 above)
Comment (AustinGitHub) [reply ytmusicapi/parsers/search.py:132]: If you check the before and after jsons I attached to the PR, it will also show this difference

File: ytmusicapi/parsers/search.py [hunk 4]
```diff
@@ -34,8 +34,12 @@ def parse_top_result(data, search_result_types):
             search_result["videoType"] = nav(on_tap, NAVIGATION_VIDEO_TYPE)
 
     if result_type in ["song", "video", "album"]:
+        on_tap = data.get("onTap")
```
Comment (sigma67) [inline ytmusicapi/parsers/search.py:37]:
```suggestion
```

**Comment by AustinGitHub** [comment]
Thanks!
